### PR TITLE
Disables ragin mages and civil war of casters from firing

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
### Why?
Wizards have been designed from the very beginning to be strong solo antagonists, with the added power of gunking rounds and such. Their many spells can be combined to have some seriously strong and stupidly powerful combinations. The problem with Dynamic's raging mages and civil war of casters is that the station was flooded with these strong antagonists that were supposed to be solo, which would inevitably make the round go to shit early. It's been almost two years since Dynamic was added and the only "balancing" the flood of wizards received was that they could not cast summon guns/magic/swords.
A list of spells that raging mages can get (that I subjectively deem very strong or very damaging to the station):
- Fireball; Creates atmospheric breaches and usually crits people when it contacts
- Mutate;  Laser eyes and hulk
- Blink; Almost untouchable, this also caused an awful "meta" where crewmembers try to use one-shot-kill weapons such as shotguns and chloral syringes because Blink makes wizards too hard to catch
- Push; Wizards can teleport machineries on another part of the station, but some machineries are far harder to move around such as cloning pods since they can't be simply unwrenched. Medbay becomes effectively non-functional while Push wizards are active.
- Mindswap; Swaps minds with the target. Becomes incredibly frustrating when multiple wizards are involved.
- Apprenticeship contracts; Each and every single ragin' mage wizard can buy contracts themselves. 
- Bundle spells; Bundles have a chance to contain a spellbook that contains some pretty unique spells, the strongest of which are summoning spares and gibbing people.

A list of artifacts:
- Staff of Change; Among options are turning people into robots and turning people into xenomorphs. Victims will require either another Staff of Change or to retrieve the original Staff of Change to turn back into humans (or robots)
- Cloak of Cloaking; Become pretty much undetectable save for a method or two (and an unintended method)? And it only costs both your hands while it's active (and slower movement speed). Usually this strategy was recently used with pitbull summoning.
- Darkness Tone Setter; This just destroys all the lights on the station but it's also incredibly cheap. 

Now multiply all of these for every wizard that appears during a round. I acknowledge that not all wizards use these powerful and/or frustrating spells.

### Aim
Wizards should return back to being regular strong solo antagonists that are provided with these tools, instead of a regularly-occurring flood of strong solo antagonists. This also makes it far easier to justify giving them cool toys and spells since that means most ragin' mages wouldn't have access to it, and frankly I can't imagine it would be fun if you were a roundstart wizard but it turned out you were a ragin' mage drafted by the Civil War of Casters so you can't use the fun toys and spells that would be specifically forbidden for ragin' mages.

:cl:
 * rscdel: Ragin' mages can no longer kick in and has been relegated to adminbus-only.

### CODE REVIEW

**Comment by twirling-jesters** [comment]
While it's nice to have it as a one-off admin special every once in a while after a particularly long round, I agree that having it in regular rotation is a bit much.

**Comment by robosperg** [comment]
:+1: to removing casters, :-1: to removing ragin

**Comment by anunknownentity03** [comment]
get it gone, but lock it behind bus only

**Comment by boy2mantwicethefam** [comment]
Sure.

**Comment by boy2mantwicethefam** [comment]
Ragin' mages is now a bus ruleset for admins to spawn when they feel like it.

File: code/datums/gamemode/dynamic/dynamic_rulesets_latejoin.dm [hunk 1]
```diff
@@ -68,7 +68,7 @@
 //        RAGIN' MAGES (LATEJOIN)           ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 //                                          //
 //////////////////////////////////////////////1.01 - Lowered weight from 3 to 2
-
+/* //Currently disabled
```
Comment (ancientpower) [inline code/datums/gamemode/dynamic/dynamic_rulesets_latejoin.dm:71]: I recommend setting the weight for the rulesets to 0 so they can still be executed by admins.

**Comment by PrimeDSS13** [comment]
I know this is a weak argument but I don't really want even less modes than we currently have.
If I was gonna be an ideas guy, I would say maybe make Civil War spawn (only) two wizards, but at the same time. Then they could duke it out on the station to see which wizard is the strongest. I feel like it would go really well with #27126 if that is merged and still leave us with a fresh wizard mode.

**Comment by boy2mantwicethefam** [comment]
@PrimeDSS13 I will likely try working on wizards and maybe a different solution on ragin' mages later, but for now I want to address obvious issues and this is the most straightforward approach. I like #27126 too, but I still think that wizard as it currently is should be a roundstart antagonist only, it has always remained as a solo powerful antagonist and the station being flooded with wizards because of Dynamic always spelled the end of the round as a result, and it was either due to the sheer damage and frustration caused by the wizards or the station miraculously fending them off but remaining perpetually in a state of mistrust and paranoia since there could always be more wizards, something also observable in Cult.

**Comment by dunsel-hordon** [comment]
I would say make it explicitly stated for casters civil war that they must hunt down the other wizard instead of engaging crew. For raging mages would it be possible to dope them up with something so they are on a timer and be literally raging mages? As in the timer runs out and either they can't cast spells for a period of time or they simply gib in a giant magical explosion?

**Comment by GreatBigFailure** [comment]
Ragin mages is fixable without removing it. Having it fire less often is reasonable but game mode removal is a bad idea until we have a bigger pool of antags. 
The spells you mentioned are definitely the issue. We should work on fixing them not removing the game mode.

**Comment by Kurfursten** [comment]
>I would say make it explicitly stated for casters civil war that they must hunt down the other wizard

It does say this, and they also get a jectie to wipe out the enemy faction

>instead of engaging crew

This is enforced friendly wizard gimmick, and it doesn't really work since people will start firing on any wizard they see

>timed gib

I feel like this encourages the worst kind of speedrunning. How fast can you fireball the supermatter so you make sure you've left your mark when your timer runs out?

**Comment by boy2mantwicethefam** [comment]
@GreatBigFailure Three problems:
- Every ruleset in the game is already at 1 weight, it can't be made to fire less often without altering threat level requirements and cost
- Nerfing spells also bites solo roundstart wizards in the ass
- [I already tried nerfing Blink](https://github.com/vgstation-coders/vgstation13/pull/25522)

@dunsel-hordon As nice as it would be, it would be infeasible because people do not want to be restricted while playing wizards, so the gamemode remains stuck as "wizardtide call the shuttle".

**Comment by SonixApache** [comment]
ragin' mages should just have less power than roundstart turbowizards

**Comment by Kurfursten** [comment]
I think there is significant confusion about what ragin' mages even is.

Ragin mages is just latejoin/midround wizard. It's no different than roundstart wizard. It's identical [except you can't buy summon gun type spells]. You are probably confused because it gets its name from the old Secret round type Ragin Mages which would send several wizards. The closest dynamic equivalent to that is CWC.

**Comment by GreatBigFailure** [comment]
>Three problems:

Thing is people have been calling for changes to most of those for a while. Fireball is just the least fun thing a wizard can do and SoC while fun in concept and even fun for the victim in some cases, often just ends someone's round if they had a gimmick going because now they're a slime and an assistant stole the staff then got spaced. 
Mindswap I think is a fine spell, but either shouldn't have an apprentice option or that apprentice should cost more. The list goes on. I'm just tired of seeing the same wiz-type over and over because there's clearly defined "win" spells.

Your blink change actually seems fine and I'm surprised people didn't like it. Maybe a bit heavy handed at just not being able to bump the cooldown at all, but still. People would probably be more accepting of it if wizard was given more escape tools that aren't blink/jaunt. Maxblink spam is really fucking unfun and costs enough that it usually makes the wizard uninteresting as well.

**Comment by boy2mantwicethefam** [comment]
Maybe I shouldn't remove as many things

**Pull request closed without merging**